src/Plugin/Action/DataCalculateValue.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\DataCalculateValue.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\rules\Core\RulesActionBase;
+
+/**
+ * Provides a 'numeric calculation' action.
+ *
+ * @Action(
+ *   id = "rules_data_calculate_value",
+ *   label = @Translation("Calculates a numeric value"),
+ *   category = @Translation("Data"),
+ *   context = {
+ *     "input_1" = @ContextDefinition("float",
+ *       label = @Translation("Input value 1"),
+ *       description = @Translation("The first input value for the calculation.")
+ *     ),
+ *     "operator" = @ContextDefinition("string",
+ *       label = @Translation("Operator"),
+ *       description = @Translation("The calculation operator.")
+ *     ),
+ *     "input_2" = @ContextDefinition("float",
+ *       label = @Translation("Input value 2"),
+ *       description = @Translation("The second input value for the calculation.")
+ *     )
+ *   },
+ *  provides = {
+ *     "result" = @ContextDefinition("float",
+ *       label = @Translation("Calculated result")
+ *     )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: Add defined operation options from Drupal 7.
```

Comment:
We need a todo also for better handling integers. Integers should be covered with the float action, but we need to ensure the result is an integer then also. As we cannot read the info of the actual context provided in any way yet, let's add a todo for handling integer context and results.